dashboard/app/models/lti_section.rb

```diff
@@ -0,0 +1,28 @@
+# == Schema Information
+#
+# Table name: lti_sections
+#
+#  id             :bigint           not null, primary key
+#  lti_course_id  :bigint           not null
+#  section_id     :integer          not null
+#  name           :string(255)
+#  lms_section_id :string(255)
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#
+# Indexes
+#
+#  index_lti_sections_on_lti_course_id  (lti_course_id)
+#  index_lti_sections_on_section_id     (section_id)
+#
+class LtiSection < ApplicationRecord
+  belongs_to :lti_course
+  belongs_to :section
+  validates :section_id, uniqueness: true
+  validates :lms_section_id, presence: true
+  after_destroy :destroy_associated_section
+
+  def destroy_associated_section
+    section.destroy
+  end
+end
```

Comment:
```suggestion
class LtiSection < ApplicationRecord
  belongs_to :lti_course
  belongs_to :section, dependent: :destroy
  validates :section_id, uniqueness: true
  validates :lms_section_id, presence: true
end
```
Would this work since there is a 1:1 relationship with section?

Reply:
Rails doesn't seem to like putting the `dependent: :destroy` on the `belongs_to` side of the relationship. I think it would probably work if we added a foreign key for `lti_section_id` into the `section` table, but I'm not sure it's worth the effort and manual migration. Thoughts?

Reply:
go ahead and skip it then.